ios/Classes/SwiftChargebeeFlutterSdkPlugin.swift

```diff
@@ -214,12 +214,28 @@ extension SKProduct {
     func toMap() -> [String: Any?] {
         let map: [String: Any?] = [
             "productId": productIdentifier,
-            "productPrice": price.description,
+            "productPrice": price,
             "productTitle": localizedTitle,
-            "currencyCode": priceLocale.currencyCode
+            "currencyCode": priceLocale.currencyCode,
+            "subscriptionPeriod": subscriptionDuration()
         ]
         return map
     }
+    func subscriptionDuration() -> String {
+        let period:String = {
+            switch self.subscriptionPeriod?.unit {
+            case .day: return "day"
+            case .week: return "week"
+            case .month: return "month"
+            case .year: return "year"
+            case .none: return ""
+            case .some(_): return ""
+            }
+        }()
+        let numOfUnits = self.subscriptionPeriod?.numberOfUnits ?? 0
+        let pluralInPeriod = numOfUnits > 1 ? "s" : ""
```

Comment:
This will only work for English. In German, we say "Monat" (=month) and "Monate" (=months).

Reply:
Correct, wouldn’t even think of other language gets change. will update the PR.